client/src/api/routes/applications.ts

```diff
@@ -0,0 +1,234 @@
+ * @throws if fails
+ */
+export const useGetApplicationFromCycle = (cycleId: string, applicationId: string): UseQueryResult<ApplicationObject> => {
+    return useQuery('getApplicationFromCycle', async () => {
+        const { data, status } = await fetcher.get<ApplicationObject | Failure>(`/application/${cycleId}/${applicationId}`);
+        if(status !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as ApplicationObject);
+    });
+};
+
+/**
+ * @description POST /application
+ * @param {string} company 
+ * @param {string} position 
+ * @param {string} location 
+ * @param  {string} jobPostUrl 
+ * @param {string} description 
+ * @returns {UseQueryResult<ApplicationObject>} the newly created application
+ * @throws if fails
+ */
+export const useCreateApplication = (company: string, position: string, location: string, jobPostUrl: string, description: string): UseQueryResult<ApplicationObject> => {
+    return useQuery('createApplication', async () => {
+        const body: Partial<ApplicationObject> = {
+            company,
+            position,
+            location,
+            jobPostUrl,
+            description
+        };
+        const { data, status } = await fetcher.post<Partial<ApplicationObject>, AxiosResponse<ApplicationObject | Failure>>('/application', body);
+        if(status !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as ApplicationObject);
+    });
+};
+
+/**
+ * @description POST /application/event/:applicationId
+ * @param {string} applicationId 
+ * @param {string} title 
+ * @param {string} date 
+ * @param {string} location 
+ * @returns {UseQueryResult<EventObject>} the newly created event
+ * @throws if fails
+ */
+export const useCreateEvent = (applicationId: string, title: string, date: string, location: string): UseQueryResult<EventObject> => {
+    return useQuery('createEvent', async () => {
+        const body: Partial<EventObject> = {
+            title,
+            date,
+            location
+        };
+        const { data, status } = await fetcher.post<Partial<EventObject>, AxiosResponse<EventObject | Failure>>(`/application/event/${applicationId}`, body);
+        if(status !== 200) throw new Error(`${(data as Failure).message}\n\n${(data as Failure).error}`);
+        return (data as EventObject);
+    });
+};
+
+/**
+ * @description POST /application/contact/:applicationId
+ * @param {string} applicationId 
+ * @param {string} name 
+ * @param {string} pronouns 
+ * @param {string} location 
+ * @param {string} phone 
+ * @param {string} email 
+ * @returns {UseQueryResult<ContactObject>} the newly created contact
+ * @throws if fails
+ */
+export const useCreateContact = (applicationId: string, name: string, pronouns: string, location: string, phone: string, email: string): UseQueryResult<ContactObject> => {
+    return useQuery('createEvent', async () => {
```

Comment:
Should this be createEvent or createContact?

Reply:
Yeah he def made a mistake; good eye

Reply:
will change